python/dgl/transforms/functional.py

```diff
@@ -3856,4 +3857,133 @@ def double_radius_node_labeling(g, src, dst):
 
     return F.tensor(z, F.int64)
 
+def shortest_dist(g, root=None, directed=True, return_paths=False):
+    r"""Compute shortest distance and paths on the given graph.
+
+    This function computes the shortest distance between all pairs of nodes
+    or between all nodes and a specified root node. Corresponding shortest
+    paths can be returned and represented with edge IDs. Note that only
+    unweighted cases are allowed.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The input graph. Must be homogeneous.
+    root : int, optional
+        The root node ID if the graph is provided with a root node.
+        Default: None. If a root node ID is provided, the function returns
+        single-source shortest paths. Otherwise, it returns all-pairs
+        shortest paths.
+    directed: bool, optional
+        A flag to indicate whether the graph is considered as directed.
+        Default: True. If False, reverse edges are considered for constructing
+        shortest paths.
+    return_paths: bool, optional
+        A flag to indicate whether to return paths. Default: False. If True,
+        return shortest paths represented with edge IDs.
+
+    Returns
+    -------
+    dist: Tensor
+        The shortest distance tensor of shape :math:`(N, N)` if root is None,
+        where :math:`N` is the number of nodes in the input graph. Or the
+        shortest distance tensor of shape :math:`(N,)` if a root node is
+        specified. The distance of unreachable pairs are filled with -1.
+    paths: Tensor, optional
```

Comment:
```suggestion
    path : Tensor, optional
```

Either you use `dists` and `paths`, or `dist` and `path` for consistency.